mesa/visualization/video_viz.py

```diff
@@ -0,0 +1,184 @@
+"""Video recording components for Mesa model visualization."""
+
+import shutil
+from collections.abc import Callable, Sequence
+from pathlib import Path
+
+import matplotlib.animation as animation
+import matplotlib.pyplot as plt
+import numpy as np
+
+import mesa
+from mesa.visualization.matplotlib_renderer import (
+    MatplotlibRenderer,
+    MeasureRendererMatplotlib,
+    SpaceRenderMatplotlib,
+)
+
+
+def make_space_component(
+    agent_portrayal: Callable | None = None,
+    propertylayer_portrayal: dict | None = None,
+    post_process: Callable | None = None,
+    **space_drawing_kwargs,
+):
+    """Create a Matplotlib-based space visualization component.
+
+    Args:
+        agent_portrayal: Function to portray agents.
+        propertylayer_portrayal: Dictionary of PropertyLayer portrayal specifications
+        post_process : a callable that will be called with the Axes instance. Allows for fine tuning plots (e.g., control ticks)
+        backend: The backend to use for rendering the space. Can be "matplotlib" or "altair".
+        space_drawing_kwargs : additional keyword arguments to be passed on to the underlying space drawer function. See
+                               the functions for drawing the various spaces for further details.
+
+    ``agent_portrayal`` is called with an agent and should return a dict. Valid fields in this dict are "color",
+    "size", "marker", and "zorder". Other field are ignored and will result in a user warning.
+
+
+    Returns:
+        SpaceRenderMatplotlib: A component for rendering the space.
+    """
+    if agent_portrayal is None:
+
+        def agent_portrayal(a):
+            return {}
+
+    return SpaceRenderMatplotlib(
+        agent_portrayal,
+        propertylayer_portrayal,
+        post_process=post_process,
+        **space_drawing_kwargs,
+    )
+
+
+def make_measure_component(
```

Comment:
In the other parts of the visualization, we shifted to just calling it a plot component; what are the reasons for using the, in my view, ambiguous label of measure here?

Reply:
Thanks for asking. I kept the impression from the old (now experimental) SolaraViz: https://github.com/projectmesa/mesa/blob/69571b401d97af68d85aa49205b0f53f494ac721/mesa/experimental/solara_viz.py#L58 and https://github.com/projectmesa/mesa/blob/69571b401d97af68d85aa49205b0f53f494ac721/mesa/experimental/solara_viz.py#L72
where there's a card titled "Space" and another titled "Measure". "Plot" seems a bit amgiguous for me because we can have a plot for space, a plot for measure, and possible plots for other stuff. I might have missed some discussions when we made the change.

Reply:
I just followed a suggestion from @Corvince in #2443. 

I agree on the ambiguity, but I contend that space is different in kind from, say, line plots, histograms, or other visualizations of data created by a model. So, for now, I am fine with separeting space from other kinds of plots. 
Since we are still hashing out and refining the visualization style, I am fine with this distinction at the moment.